node/src/components/contract_runtime.rs

```diff
@@ -529,9 +529,6 @@ where
                                 engine_state.upgrade(),
                             ) {
                                 (Some(block_at_height), Some(engine_state))
-                                    // TODO: This is a band-aid. If a block is not executed, we
-                                    // should fetch old blocks backwards until we find one with a
-                                    // state root that exists and execute forwards.
```

Comment:
@henrytill @dwerner So while I see some failure modes, I am not sure how to handle them in the long run, so I'm removing this comment.